pkg/authorization/api/helpers.go

```diff
@@ -13,7 +13,28 @@ import (
 	// uservalidation "github.com/openshift/origin/pkg/user/api/validation"
 )
 
-func ExpandResources(rawResources sets.String) sets.String {
+// NormalizeResources expands all resource groups and forces all resources to lower case.
+// If the rawResources are already normalized, it returns the original set to avoid the
+// allocation and GC cost, since this is hit multiple times for every REST call.
+// That means you should NEVER MODIFY THE RESULT of this call.
```

Comment:
clarify you can get the original back... if you were comfortable modifying the original, you can modify the result